lib/archethic/oracle_chain/services/uco_price.ex

```diff
@@ -13,20 +13,59 @@ defmodule Archethic.OracleChain.Services.UCOPrice do
 
   @impl Impl
   @spec fetch() :: {:ok, %{required(String.t()) => any()}} | {:error, any()}
-  def fetch, do: provider().fetch(@pairs)
+  def fetch do
+    ## retrieve prices from configured providers and filter results marked as errors
+    prices =
+      providers()
+      |> Task.async_stream(fn provider ->
+        case provider.fetch(@pairs) do
+          {:ok, _prices} = result ->
+            result
+
+          {:error, reason} ->
+            Logger.warning("Service : #{inspect(__MODULE__)} : Cannot fetch values from
+                provider: #{inspect(provider)} with reason : #{inspect(reason)}.")
+
+            {false, provider}
+        end
+      end)
+      |> Enum.map(fn
```

Comment:
I think you have to handle the timeout and not crash the tasks if it's happening.
You can use the option: `on_timeout: :kill_task`